treeherder/config/settings.py

```diff
@@ -196,6 +215,56 @@
     }
 }
 
+# SECURITY
+USE_X_FORWARDED_HOST = True
+USE_X_FORWARDED_PORT = True
+SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTO', 'https')
+
+if SITE_URL.startswith('https://'):
+    SECURE_SSL_REDIRECT = True
+    SECURE_HSTS_INCLUDE_SUBDOMAINS = True
+    SECURE_HSTS_PRELOAD = True
+    SECURE_HSTS_SECONDS = int(timedelta(days=365).total_seconds())
+    # Mark session and CSRF cookies as being HTTPS-only.
+    CSRF_COOKIE_SECURE = True
+    SESSION_COOKIE_SECURE = True
+
+SECURE_CONTENT_TYPE_NOSNIFF = True  # Set the `X-Content-Type-Options` header to `nosniff`.
+SECURE_BROWSER_XSS_FILTER = True  # Sets the `X-XSS-Protection` header.
+
+# System Checks
+SILENCED_SYSTEM_CHECKS = [
+    # We can't set CSRF_COOKIE_HTTPONLY to True since the requests to the API
+    # made using Angular's `httpProvider` require access to the cookie.
+    'security.W017',
+    # We can't set X_FRAME_OPTIONS to DENY since renewal of auth token requires
+    # opening an invisible iframe with the same origin.
+    'security.W019'
+]
+
+# User Agents
+# User agents which will be blocked from making requests to the site.
+# Note: we use the default setting for X_FRAME_OPTIONS ('SAMEORIGIN') so Auth0
```

Comment:
This comment need to be slightly higher up, above the `security.W019` line and combined with the comment there.

Reply:
Ah, sorry I misread your last comment!

Reply:
Oh I actually wrote the wrong thing haha

Reply:
Rebased into the latest commit

Reply:
👍